aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -504,6 +504,13 @@ static void ndtri_kernel(TensorIteratorBase& iter) {
       });
 }
 
+static void log_ndtr_kernel(TensorIteratorBase& iter) {
+  TORCH_INTERNAL_ASSERT(iter.ntensors() == 2);
```

Comment:
Interesting that we have all these internal asserts for the number of tensors now. I wonder where that came from?

Reply:
Indeed, I think this assert only provides marginal benefit. 
This iter instance is created by calling TensorIterator::unary_op(result, self), which is guaranteed to end up having 2 operands. 

How do we think of such defensive programming practice? Shall we remove all such tests in this file? Or swap it with TORCH_INTERNAL_ASSERT_DEBUG_ONLY? Or leave it :) ?

Reply:
Why were they added?

Reply:
As far as I can trace, this assert is first introduced 3 years ago... and it's a result of replacing "AT_ASSERT(ntensor == 2);" Back then, the inputs are passed in as array pointers with count: (int ntensors, char** data_). So it made sense to do such assert at that time. 

I think this line of assert is no longer needed. Just like BinaryOpKernels.cpp, these is no need to assert ntensors() == 3. 

See https://github.com/pytorch/pytorch/pull/21475/files#diff-ccaf9206c0f650871f27d32b168c31bf5706d18ee1b9c8210412cc98367ab0cdR190 
for first introduction of this line.

Reply:
I agree; we could file a better engineering to follow-up and review (and likely remove) these asserts